src/main/scala/rps/Game.scala

```diff
@@ -1,33 +1,55 @@
 package rps
 import scala.io.StdIn.readLine
 
+sealed abstract class Move {
+  def toString : String
+}
+case object ROCK extends Move {
+  override def toString = "Rock"
+}
+case object PAPER extends Move {
+  override def toString = "Paper"
+}
+case object SCISSORS extends Move {
+  override def toString = "Scissors"
+}
+case object INVALID extends Move {
+  override def toString = "Invalid Move!"
+}
+
 object Game {
   val randomGen = scala.util.Random
 
-  def moveRepr(move: Int): String = {
-    move match {
-      case 0 => "Rock"
-      case 1 => "Paper"
-      case 2 => "Scissors"
-      case _ => "Invalid move!"
-    }
+  implicit def intToMove(move: Option[Int]): Move = {
```

Comment:
don't make this implicit. There's little gain here and potentially some confusion.

Reply:
also ideally `intToMove` should take `Int` as an input as opposed to `Option[Int]`